main.go

```diff
@@ -43,42 +47,71 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
-func main() {
-	var metricsAddr string
-	var enableLeaderElection bool
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
+var flags = struct {
+	EnableLeaderElection bool
+	MetricsAddr          string
+}{}
+
+func initFlags() {
+	flag.BoolVar(&flags.EnableLeaderElection, "enable-leader-election", false, "Enable leader election for controller manager.")
+	flag.StringVar(&flags.MetricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
```

Comment:
Probably won't matter for us right now, but FYI: These will change to `leader-elect` and `metrics-bind-addr` in the "future". See: https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/a16ac69cdb04f419570ef4076139edfb8524bb38#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261

Reply:
It's like that in the master branch: https://github.com/kubernetes-sigs/kubebuilder/blob/faff1482f454cabffb4e685a138a72a3ca040324/testdata/project-v3/main.go#L55-L57

For now I'm using kubebuilder 2.4.2 to avoid friction. 3.0.0 is still in beta.

Reply:
https://github.com/giantswarm/capi-migration/pull/15